### PR TITLE
Move to static

### PR DESCRIPTION
**Pull request by rahul-daga-imgtec**

### CODE REVIEW

File: lwm2m-client-ipso-digital-input.c [hunk 1]
```diff
@@ -41,404 +41,262 @@
...
 typedef struct
 {
-	bool State;
-	int64_t Counter;
-	bool Polarity;
-	int64_t DebouncePeriod;
-	int64_t EdgeSelection;
-	char ApplicationType[MAX_STR_SIZE];
-	char SensoryType[MAX_STR_SIZE];
+    bool State;
```
Comment (seank-img) [inline lwm2m-client-ipso-digital-input.c]: Should use AwaBoolean

File: lwm2m-client-ipso-digital-input.c [hunk 2]
```diff
@@ -41,404 +41,262 @@
...
+        printf("Failed to define SensoryType resource\n");
+        return 1;
+    }
+
+    error = AwaStaticClient_DefineResourceWithHandler(awaClient, "CounterReset", IPSO_DIGITAL_INPUT_OBJECT , IPSO_DIGITAL_INPUT_COUNTER_RESET, AwaResourceType_Opaque, 0, 1, AwaResourceOperations_Execute, digitalInputHandler);
+    if (error != AwaError_Success)
+    {
+        printf("Failed to define CounterReset resource\n");
+        return 1;
+    }
```
Comment (seank-img) [inline lwm2m-client-ipso-digital-input.c]: The IPSO digital input objects and their resources still need to be created after they are defined:

    for(int i = 0; i < DIGITAL_INPUTS; i++)
    {
        AwaStaticClient_CreateObject(awaClient, IPSO_DIGITAL_INPUT_OBJECT, i); //mandatory resources created here
        AwaStaticClient_CreateResource(awaClient, IPSO_DIGITAL_INPUT_OBJECT, i, IPSO_DIGITAL_INPUT_COUNTER);
    }

File: lwm2m-client-ipso-digital-input.c [hunk 3]
```diff
@@ -41,404 +41,262 @@
...
-	return 0;
+    AwaError error;
+    error = AwaStaticClient_DefineObjectWithHandler(awaClient, "DigitalInput", IPSO_DIGITAL_INPUT_OBJECT, 0, DIGITAL_INPUTS, digitalInputHandler);
+    if (error != AwaError_Success)
+    {
+        printf("Failed to register flow access object\n");
+        return 1;
+    }
+
+    error = AwaStaticClient_DefineResourceWithHandler(awaClient, "State", IPSO_DIGITAL_INPUT_OBJECT, IPSO_DIGITAL_INPUT_STATE, AwaResourceType_Boolean, 0, 1, AwaResourceOperations_ReadOnly, digitalInputHandler);
```
Comment (seank-img) [inline lwm2m-client-ipso-digital-input.c]: IPSO Smart Objects 1.0 spec says this should be mandatory - change minimumInstances to 1:

    error = AwaStaticClient_DefineResourceWithHandler(awaClient, "State", IPSO_DIGITAL_INPUT_OBJECT, IPSO_DIGITAL_INPUT_STATE, AwaResourceType_Boolean, 1, 1, AwaResourceOperations_ReadOnly, digitalInputHandler);

The digital input state state doesn't toggle with button press but to follow the spec the resource should be created. I'm not sure if there is a way to get down/up events with the button sensor in contiki, but if there is a way this resource should be updated with the button state.

File: lwm2m-client-ipso-digital-input.c [hunk 4]
```diff
@@ -41,404 +41,262 @@
...
+                    if(*dataSize < sizeof(digitalInputs[objectInstanceID].ApplicationType))
+                    {
+                        memcpy(digitalInputs[objectInstanceID].ApplicationType, *dataPointer, *dataSize);
+                        digitalInputs[objectInstanceID].ApplicationType[*dataSize] = '\0';
+                    }
+                    else
+                    {
+                        result = AwaResult_BadRequest;
+                    }
+                break;
```
Comment (pratik-prajapati-imgtec) [inline lwm2m-client-ipso-digital-input.c]: indentation

File: lwm2m-client-flow-access-object.c [hunk 5]
```diff
@@ -56,281 +57,229 @@
...
-	REGISTER_FLOW_ACCESS_OBJECT_RESOURCE(context, "RememberMeToken", \
-		FLOWM2M_FLOW_ACCESS_OBJECT_REMEMBERMETOKEN, ResourceTypeEnum_TypeString);
-	REGISTER_FLOW_ACCESS_OBJECT_RESOURCE(context, "RememberMeTokenExpiry", \
-		FLOWM2M_FLOW_ACCESS_OBJECT_REMEMBERMETOKENEXPIRY, ResourceTypeEnum_TypeOpaque);
+	AwaError error;
 
+	error = AwaStaticClient_CreateObjectInstance(awaClient, FLOWM2M_FLOW_ACCESS_OBJECT, 0);
+	if (error != AwaError_Success)
+	{
+		printf("failed to create flow access object instance\n");
```
Comment (abhijit-mahajani-imgtec) [inline lwm2m-client-flow-access-object.c]: Make first letter of any debug print Capital to maintain the consistency.
failed -> Failed

File: lwm2m-client-flow-object.c [hunk 6]
```diff
@@ -221,28 +226,33 @@ AwaResult handler(AwaStaticClient *client, AwaOperation operation, AwaObjectID o
...
 				case 2:
-					snprintf(flow[objectInstanceID].DeviceType, *dataSize, "%s", (char *)*dataPointer);
+					memcpy(flow[objectInstanceID].DeviceType, *dataPointer, *dataSize);
+					flow[objectInstanceID].DeviceType[*dataSize] = '\0';
 					*changed = true;
 					break;
 
 				case 3:
-					snprintf(flow[objectInstanceID].Name, *dataSize, "%s", (char *)*dataPointer);
+				    memcpy(flow[objectInstanceID].Name, *dataPointer, *dataSize);
```
Comment (abhijit-mahajani-imgtec) [inline lwm2m-client-flow-object.c]: check indentation

File: lwm2m-client-flow-object.c [hunk 7]
```diff
@@ -221,28 +226,33 @@ AwaResult handler(AwaStaticClient *client, AwaOperation operation, AwaObjectID o
...
 				case 4:
-					snprintf(flow[objectInstanceID].Description, *dataSize, "%s", (char *)*dataPointer);
+					memcpy(flow[objectInstanceID].Description,  *dataPointer, *dataSize);
+					flow[objectInstanceID].Description[*dataSize] = '\0';
 					*changed = true;
 					break;
 
 				case 5:
-					snprintf(flow[objectInstanceID].FCAP, *dataSize, "%s", (char *)*dataPointer);
+				    memcpy(flow[objectInstanceID].FCAP, *dataPointer, *dataSize);
```
Comment (abhijit-mahajani-imgtec) [inline lwm2m-client-flow-object.c]: check indentation

**Comment by abhijit-mahajani-imgtec** [comment]
https://github.com/CreatorKit/libobjects/pull/6/commits/893b873f83c039914b39bef4c23090bf7cdedd16

Fix the typo in the commit message

**Comment by abhijit-mahajani-imgtec** [comment]
@rahul-daga-imgtec , please update manifest file to pull Awa 0.1.7, else verification of this static apis is not possible.

**Comment by abhijit-mahajani-imgtec** [comment]
Approved

**Comment by nikhil-zinjurde-imgtec** [comment]
Verified failed, I got an error while doing 'make contiki' with these changes in libobjects
````
I got an error while building the static API changes in libobjects repo
../../packages/libobjects/lwm2m-client-flow-object.c:102:1: warning: missing braces around initializer
../../packages/libobjects/lwm2m-client-flow-object.c:102:1: warning: (near initialization for 'flow[0]')
../../packages/libobjects/lwm2m-client-flow-object.c:123:11: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'handler'
../../packages/libobjects/lwm2m-client-flow-object.c: In function 'DefineFlowObject':
../../packages/libobjects/lwm2m-client-flow-object.c:311:5: warning: implicit declaration of function 'AwaStaticClient_DefineObjectWithHandler'
../../packages/libobjects/lwm2m-client-flow-object.c:311:112: error: 'handler' undeclared (first use in this function)
../../packages/libobjects/lwm2m-client-flow-object.c:311:112: note: each undeclared identifier is reported only once for each function it appears in
../../packages/libobjects/lwm2m-client-flow-object.c:318:5: warning: implicit declaration of function 'AwaStaticClient_DefineResourceWithHandler'
../../packages/libobjects/lwm2m-client-flow-object.c: At top level:
../../packages/libobjects/lwm2m-client-flow-object.c:77:13: warning: 'licenseeSecret' defined but not used
../../packages/libobjects/lwm2m-client-flow-object.c:102:19: warning: 'flow' defined but not used
../../packages/libobjects/lwm2m-client-flow-object.c:104:13: warning: 'CalculateLicenseeHash' defined but not used
make[1]: *** [obj_mikro-e/lwm2m-client-flow-object.o] Error 255
make[1]: Leaving directory `/home/nikhil.zinjurde/repository/test-pull/packages/led-actuator'
make: *** [build_contiki] Error 2
````

**Comment by nikhil-zinjurde-imgtec** [comment]
QA verified

**Pull request merged**